### ISSUE TITLE
Example not working: 12-graphql-apollo

### ISSUE THREAD

**Issue by Decimator** [nestjs/nest#1332]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
The vanilla "12-graphql-apollo" example can't be built. When attempting to build and then run the built files, this error appears:
"Query.getCats defined in resolvers, but not in schema"

## Expected behavior
The vanilla example should build correctly using tsc.

## Minimal reproduction of the problem with instructions

1. Go to 'nest -> sample -> 12-graphql-apollo' directory (the example project)
2. $ npm install
3. $ npm run prestart:prod
4. $ cd dist
5. $ node main.js
6. Server starts, then crashes with the following error log:

`(node:26488) UnhandledPromiseRejectionWarning: Error: Query.getCats defined in resolvers, but not in schema`
`(node:26488) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)`
`(node:26488) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

## What is the motivation / use case for changing the behavior?
To help NestJS newcomers to understand and be able to use NestJS.

## Environment

<pre><code>
Nest version: 5.4.1
 
For Tooling issues:
- Node version: 8.11.4
- Platform:  Windows

Others:

</code></pre>

**Comment by Decimator** [comment]
I've spent the weekend trying to find a way around this, no success though unfortunately. I'm new to NestJS and TypeScript; my apologies if there's an obvious fix.

My current theory is that tsc isn't compiling the GraphQL schemas properly? I noticed that within the dev src files, the graphql.schema.ts file is being automatically updated by ts-node, however after tsc runs, the graphql.schema.js file in the /dist folder appears to be empty. It contains the correct exports, but the exported classes are completely empty?

Built version of graphql.schema.js:
`"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class CreateCatInput {
}
exports.CreateCatInput = CreateCatInput;
class Cat {
}
exports.Cat = Cat;
class IMutation {
}
exports.IMutation = IMutation;
class IQuery {
}
exports.IQuery = IQuery;
class ISubscription {
}
exports.ISubscription = ISubscription;
//# sourceMappingURL=graphql.schema.js.map`

That seems like it isn't compiling correctly?

**Comment by brianmcd** [comment]
The main issue is that  `dist` isn't getting `cats/cats.graphql` copied over into it.  The example works if you run `node dist/main.js`, but not if you run `node main.js` from inside dist.

The `ApplicationModule` in the cats example has the `typePaths` set like this:
https://github.com/nestjs/nest/blob/c6ec8878e0864469fa5e83d97d3ba07df397b09e/sample/12-graphql-apollo/src/app.module.ts#L10

Since those `.graphql` files aren't getting copied to dist, running `node main.js` inside `dist/` doesn't see them.  When you run `node dist/main.js`, it's able to find them in the `src/` directory.  I think in a real app you'd either copy those `.graphql` files into `dist` or use a different mechanism to load them yourself and pass them to the `GraphQLModule` using `typeDefs` instead of `typePaths` (assuming that's how those options work - never used them myself).

Also, just to mention it since you may not be aware, you can do `npm run start:prod`, which should run the example as expected.  npm will run the `prestart:prod` task automatically before `start:prod`.

**Comment by Decimator** [comment]
Thank you for the help @brianmcd! That's really helpful, and does indeed work.

So to conclude, it looks like I was asking two separate questions (both which @brianmcd generously answered):

**Question:** When the [Minimal reproduction of the problem] commands are run, there seems to be an error that prevents the example from working.
**Answer:** The example does work, when run with the correct command $npm run start:prod.

**Question:** When building the project into /dist, node doesn't seem to be able to find the GraphQL schemas. Specifically it generates the error "Error: Query.getCats defined in resolvers, but not in schema"
**Answer:** The .graphql files in the /src directory are not (currently) being copied over to /dist when the application is built.
**Solution:** I tested Brian's proposed solution by manually copying the "cats.graphql" file from /src/cats, into the /dist folder (right alongside main.js) after building ($npm prestart:prod). Then when I ran the command $node main.js from within the /dist folder, and it works! This is potentially important for certain deployment options (in my case, deploying to an AWS Elastic Beanstalk instance. I have the project setup to deploy only the contents of the /dist directory)
